### PR TITLE
{lib}[GCCcore/10.3.0] tbb v2021.2.0

### PR DESCRIPTION
**Pull request by robert-mijakovic**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
nuc.lan - Linux Fedora 33, x86_64, Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz, Python 3.9.5
See https://gist.github.com/08e5644082539c4cca21eb26619de6f5 for a full test report.

**Comment by verdurin** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/t/tbb/tbb-2021.2.0-GCCcore-10.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+             'b145cf83d71b7184567ee477c233ba2393977f5c4cc5db6c2afaed6ab9d0f5ef']
+
+builddependencies = [
+    ('binutils', '2.36.1'),
+    ('CMake', '3.20.1'),
+]
+
+sanity_check_paths = {
+    'files': ['include/%(namelower)s/tbb.h'] +
+             ['lib64/lib%(namelower)s.so'],
```
Comment (verdurin) [inline easybuild/easyconfigs/t/tbb/tbb-2021.2.0-GCCcore-10.3.0.eb:28]: @robert-mijakovic please use the `SHLIB_EXT` macro for shared libraries.

File: easybuild/easyconfigs/t/tbb/tbb-2021.2.0_relaxed.patch [hunk 2]
```diff
@@ -0,0 +1,48 @@
+--- include/oneapi/tbb/detail/_concurrent_queue_base.h	2021-01-27 21:03:49.000000000 +0100
```
Comment (verdurin) [inline easybuild/easyconfigs/t/tbb/tbb-2021.2.0_relaxed.patch:1]: @robert-mijakovic please include a comment explaining the purpose of the patch

**Review by verdurin — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/t/tbb/tbb-2021.2.0_relaxed.patch (hunk 2 above)
Comment (robert-mijakovic) [reply easybuild/easyconfigs/t/tbb/tbb-2021.2.0_relaxed.patch:1]: To be honest I remember I had some issue compiling it and then I introduced this patch but don't remember the reason. I have just triggered the build but couldn't reproduce it. Maybe it was affecting the previous release of tbb.

**Review by verdurin — APPROVED** [review]

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
nuc.lan - Linux Fedora 33, x86_64, Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz, Python 3.9.5
See https://gist.github.com/557e283f75913cc0bbd73f1fb01b02b2 for a full test report.

**Comment by verdurin** [comment]
@boegelbot please test @ generoso

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
easybuild-c7.novalocal - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/7baf5dd361213ab068491436623d37f2 for a full test report.

File: easybuild/easyconfigs/t/tbb/tbb-2021.2.0_relaxed.patch (hunk 2 above)
Comment (robert-mijakovic) [reply easybuild/easyconfigs/t/tbb/tbb-2021.2.0_relaxed.patch]: I think the issue you experienced while testing is the one I faced. I will reenable the patch,  put the comment in EC and ask you to retrigger your test.

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
nuc.lan - Linux Fedora 33, x86_64, Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz, Python 3.9.5
See https://gist.github.com/a6d5f6aac242cda0eea4051d6482497c for a full test report.

File: easybuild/easyconfigs/t/tbb/tbb-2021.2.0_relaxed.patch (hunk 2 above)
Comment (verdurin) [reply easybuild/easyconfigs/t/tbb/tbb-2021.2.0_relaxed.patch]: @robert-mijakovic still seeing this build error:
```
[ 76%] Building CXX object test/CMakeFiles/conformance_concurrent_vector.dir/conformance/conformance_concurrent_vector.cpp.o
cd /mnt/ds918plus/eb/nuc-fedora32/build/tbb/2021.2.0/GCCcore-10.3.0/easybuild_obj/test && /mnt/ds918plus/eb/nuc-fedora32/software/GCCcore/10.3.0/bin/g++  -I/mnt/ds918plus/eb/nuc-fedora32/build/tbb/2021.2.0/GCCcore-10.3.0/oneTBB-2021.2.0/test/.. -I/mnt/ds918plus/eb/nuc-f
edora32/build/tbb/2021.2.0/GCCcore-10.3.0/oneTBB-2021.2.0/test -I/mnt/ds918plus/eb/nuc-fedora32/build/tbb/2021.2.0/GCCcore-10.3.0/oneTBB-2021.2.0/src/tbb/../../include -O2 -ftree-vectorize -march=native -fno-math-errno -O3 -DNDEBUG -Wall -Wextra -Werror -Wfatal-errors -
Wshadow -Wcast-qual -Woverloaded-virtual -Wnon-virtual-dtor -mrtm -pthread -std=c++11 -MD -MT test/CMakeFiles/conformance_concurrent_vector.dir/conformance/conformance_concurrent_vector.cpp.o -MF CMakeFiles/conformance_concurrent_vector.dir/conformance/conformance_concu
rrent_vector.cpp.o.d -o CMakeFiles/conformance_concurrent_vector.dir/conformance/conformance_concurrent_vector.cpp.o -c /mnt/ds918plus/eb/nuc-fedora32/build/tbb/2021.2.0/GCCcore-10.3.0/oneTBB-2021.2.0/test/conformance/conformance_concurrent_vector.cpp
In file included from /mnt/ds918plus/eb/nuc-fedora32/build/tbb/2021.2.0/GCCcore-10.3.0/oneTBB-2021.2.0/src/tbb/../../include/oneapi/tbb/concurrent_queue.h:21,
                 from /mnt/ds918plus/eb/nuc-fedora32/build/tbb/2021.2.0/GCCcore-10.3.0/oneTBB-2021.2.0/test/conformance/conformance_concurrent_queue.cpp:24:
/mnt/ds918plus/eb/nuc-fedora32/build/tbb/2021.2.0/GCCcore-10.3.0/oneTBB-2021.2.0/src/tbb/../../include/oneapi/tbb/detail/_concurrent_queue_base.h: In function void TestConstructors() [with CQ = ConcQWithSizeWrapper<Bar>; T = Bar; TIter = BarIterator; CQ_EX = ConcQWithSi
zeWrapper<BarEx>; T_EX = BarEx]:
/mnt/ds918plus/eb/nuc-fedora32/build/tbb/2021.2.0/GCCcore-10.3.0/oneTBB-2021.2.0/src/tbb/../../include/oneapi/tbb/detail/_concurrent_queue_base.h:547:29: error: dqb_ex.tbb::detail::d1::concurrent_queue_iterator_base<BarEx, tbb::detail::d1::cache_aligned_allocator<BarEx>
 >::my_array[7] may be used uninitialized in this function [-Werror=maybe-uninitialized]
  547 |                 my_array[i] = other.my_array[i];
      |                 ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
cc1plus: all warnings being treated as errors
make[2]: *** [test/CMakeFiles/conformance_concurrent_queue.dir/build.make:79: test/CMakeFiles/conformance_concurrent_queue.dir/conformance/conformance_concurrent_queue.cpp.o] Error 1
make[2]: Leaving directory '/mnt/ds918plus/eb/nuc-fedora32/build/tbb/2021.2.0/GCCcore-10.3.0/easybuild_obj'
make[1]: *** [CMakeFiles/Makefile2:3085: test/CMakeFiles/conformance_concurrent_queue.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

Comment (robert-mijakovic) [reply easybuild/easyconfigs/t/tbb/tbb-2021.2.0_relaxed.patch:1]: I have just reenabled the patch. Sorry for the slowness.

Comment (robert-mijakovic) [reply easybuild/easyconfigs/t/tbb/tbb-2021.2.0_relaxed.patch:1]: @verdurin May I ask you how do you create a test report?

Comment (verdurin) [reply easybuild/easyconfigs/t/tbb/tbb-2021.2.0_relaxed.patch:1]: @robert-mijakovic if you have the Github integration already configured, it's pretty simple. See https://docs.easybuild.io/en/latest/Integration_with_GitHub.html#uploading-test-reports-upload-test-report

**Comment by verdurin** [comment]
@boegelbot please test @ generoso

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
nuc.lan - Linux Fedora 33, x86_64, Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz, Python 3.9.5
See https://gist.github.com/59d6329bd62987a065218b5472022744 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3147.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/249a78fbc48a9a69884c6e7e67e9ff9f for a full test report.

**Comment by boegel** [comment]
We already have `tbb-2020.3-GCCcore-10.3.0.eb` which is used as a dependency for esyconfigs using `foss/2021a`, so I'll update #13361 to use that

**Pull request closed without merging**